File: tests/fixtures/ent/basic/svc.go
package main

import (
	"context"

	"ent-fixture/ent"
)

func createUser(ctx context.Context, client *ent.Client) error {
	_, err := client.User.Create().SetName("alice").SetAge(30).Save(ctx)
	return err
}

func listUsers(ctx context.Context, client *ent.Client) error {
	_, err := client.User.Query().All(ctx)
	return err
}

func getUser(ctx context.Context, client *ent.Client) error {
	_, err := client.User.Get(ctx, 1)
	return err
}

func updateUser(ctx context.Context, client *ent.Client) error {
	_, err := client.User.Update().SetAge(31).Save(ctx)
	return err
}

func deleteUser(ctx context.Context, client *ent.Client) error {
	_, err := client.User.Delete().Exec(ctx)
	return err
}

func createOrder(ctx context.Context, client *ent.Client) error {
	_, err := client.Order.Create().SetTotal(100).Save(ctx)
	return err
}

func queryOrders(ctx context.Context, client *ent.Client) error {
	_, err := client.Order.Query().All(ctx)
	return err
}

File: tests/fixtures/tortoise/basic/svc.py
"""tortoise-orm CRUD surface."""

from tortoise import fields
from tortoise.models import Model


class User(Model):
    id = fields.IntField(pk=True)
    name = fields.CharField(max_length=100)
    age = fields.IntField()


class Order(Model):
    id = fields.IntField(pk=True)
    total = fields.IntField()


async def create_user():
    return await User.create(name='alice', age=30)


async def list_users():
    return await User.all()


async def filter_users():
    return await User.filter(name='alice').all()


async def get_user(uid: int):
    return await User.get(id=uid)


async def get_or_none_user(uid: int):
    return await User.get_or_none(id=uid)


async def update_user():
    return await User.filter(id=1).update(age=31)


async def delete_user():
    return await User.filter(id=1).delete()


async def count_users():
    return await User.all().count()


async def create_order(total: int):
    return await Order.create(total=total)


async def filter_orders():
    return await Order.filter(total__gt=100).all()
